### PR TITLE
fixing continuous partial derivatives for fmi

### PR DESCRIPTION
**Pull request by wibraun**
- rewrite the structure for the dependencies and the partial derivatives generation
- enable them by default
- adjust C-runtime code generation

### CODE REVIEW

**Comment by OpenModelica-Hudson** [comment]
The job failed the coding style; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-07-05_16-55-16](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4376/console). Remember to clean up the commit you did that breaks the coding style rather than piling a new commit on top of the old ones. This goes double for accidental addition of files that you need to remove.

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-07-05_17-09-16](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4377/).

File: Compiler/BackEnd/SymbolicJacobian.mo [hunk 1]
```diff
@@ -1579,6 +1579,85 @@ algorithm
...
+  BackendDAE.SparseColoring sparseColoring;
+
+  DAE.FunctionTree funcs, functionTree;
+
+  BackendDAE.ExtraInfo ei;
+  FCore.Cache cache;
+  FCore.Graph graph;
+algorithm
+try
+	// for now perform on collapsed system
```
Comment (lochel) [inline Compiler/BackEnd/SymbolicJacobian.mo]: Please, don't push tabs.

File: Compiler/SimCode/SimCodeMain.mo [hunk 2]
```diff
@@ -143,14 +145,22 @@ protected
...
   (libs,libPaths,includes, includeDirs, recordDecls, functions, literals) :=
     SimCodeUtil.createFunctions(p, inBackendDAE);
+
   simCode := createSimCode(inBackendDAE, inInitDAE, NONE(), inRemovedInitialEquationLst,
     className, filenamePrefix, fileDir, functions, includes, includeDirs, libs, libPaths, p, SOME(simSettings), recordDecls, literals, Absyn.FUNCTIONARGS({},{}), isFMU=true, FMUVersion=FMUVersion);
+
+  if FMI.isFMIVersion20(FMUVersion) then
+    modelStructure := SimCodeUtil.createFMIModelStructure(inFMIDer, simCode.modelInfo, simCode.crefToSimVarHT);
+    simCode.modelStructure := modelStructure;
+  end if;
```
Comment (lochel) [inline Compiler/SimCode/SimCodeMain.mo]: Why did you move this out of the function `createSimCode`?

Comment (wibraun) [inline Compiler/SimCode/SimCodeMain.mo]: Because createSimCode is ways too huge and a mass, it demands a clean up and this a a first step ;)

File: Compiler/BackEnd/SymbolicJacobian.mo (hunk 1 above)
Comment (wibraun) [inline Compiler/BackEnd/SymbolicJacobian.mo]: Isn't there a Hudson job to watch therefore?

File: Compiler/SimCode/SimCodeMain.mo (hunk 2 above)
Comment (lochel) [inline Compiler/SimCode/SimCodeMain.mo]: Sounds like a good idea. However, wouldn't it be better to wrap up the parts that belong together instead of spreading them around?

File: Compiler/BackEnd/SymbolicJacobian.mo (hunk 1 above)
Comment (sjoelund) [inline Compiler/BackEnd/SymbolicJacobian.mo]: Not since we added the pre-commit hook that replaces all tabs. You shouldn't disable it ;)

Comment (wibraun) [inline Compiler/BackEnd/SymbolicJacobian.mo]: I'm not aware of disabled it...

File: Compiler/SimCode/SimCodeMain.mo (hunk 2 above)
Comment (wibraun) [inline Compiler/SimCode/SimCodeMain.mo]: This is not spreading around, since in my point of view this is an fmi part, so I put it to the` function generateModelCodeFMU` and removed it from the general `createSimCode`. Actually I would like to see most parts of `SimCode.SimCode` record creatable with single functions, instead by one big function like createSimCode.

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-07-12_15-29-51](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4397/).

**Comment by lochel** [comment]
@wibraun What is the status of this PR and [ticket:4424](https://trac.openmodelica.org/OpenModelica/ticket/4424)?

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-02_11-02-22](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4603/).

**Comment by OpenModelica-Hudson** [comment]
The job failed to compile; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-10-02_13-55-36](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4610/console).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-02_14-09-10](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4611/).

**Comment by OpenModelica-Hudson** [comment]
Wrong hash OMCompiler 32afee5bc15a309b195d97bf51fd2e1c65f04364 8755be492728e234d204f75ec2cfd0eba763a8fc

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-04_12-24-37](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4624/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-04_14-32-44](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4628/).

**Comment by OpenModelica-Hudson** [comment]
The job failed the coding style; for details, see [OpenModelica_TEST_PULL_REQUEST 2017-10-04_21-33-45](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4634/console). Remember to clean up the commit you did that breaks the coding style rather than piling a new commit on top of the old ones. This goes double for accidental addition of files that you need to remove.

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-04_21-38-17](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4635/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-06_02-14-43](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4639/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-09_15-20-48](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4661/).

**Comment by OpenModelica-Hudson** [comment]
Wrong hash OMCompiler b6fad2c1298a6a0c2b5d84006a89f17d4f213f33 4785bf895825b559504dae9a85b60d6042da096e

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-10_16-29-27](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4694/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-12_02-10-33](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4717/).

**Comment by OpenModelica-Hudson** [comment]
The test suite is unstable according to [OpenModelica_TEST_PULL_REQUEST 2017-10-16_12-27-28](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4751/).

**Comment by OpenModelica-Hudson** [comment]
The tests run correctly according to [OpenModelica_TEST_PULL_REQUEST 2017-10-16_14-07-56](https://test.openmodelica.org/hudson//job/OpenModelica_TEST_PULL_REQUEST/4753/console).
